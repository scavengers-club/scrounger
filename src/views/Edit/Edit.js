import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import AdventureForm from '../../components/AdventureForm/AdventureForm';
import { deleteAdventureById, editAdventure, getAdventureById } from '../../services/adventures';

export default function Edit({ currentUser }) {
  const [adventure, setAdventure] = useState({});
  const userId = currentUser.id;
  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await getAdventureById(params.id);
      setAdventure(data);
    };
    fetchData();
  }, [params.id]);

  const updateAdventure = (key, value) => {
    adventure[key] = value;
    setAdventure({ ...adventure, user_id: userId });
  };

  const deleteAdventure = async () => {
    try {
      await deleteAdventureById(params.id);
      alert('adventure deleted');
    } catch {
      alert('something went wrong!');
    }
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await editAdventure(adventure);
      confirm('are you ready to submit this adventure?');
      //history.push(`/user/${currentUser.id}`);
    } catch {
      alert('something went wrong!');
    }
  };

  return (
    <div>
      <AdventureForm {...adventure} {...{ updateAdventure, handleSubmit, deleteAdventure }} />
    </div>
  );
}
