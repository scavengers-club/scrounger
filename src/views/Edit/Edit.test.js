import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router';
import Edit from './Edit';

test('Edit should render prefilled form ', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/adventure/1/edit']}>
      <Route path="/adventure/:id/edit">
        <Edit
          currentUser={{ id: 'cdac1946-18d3-46a7-b531-9a2fb66ade51' }}
          match={{ params: { id: 1 } }}
        />{' '}
      </Route>
    </MemoryRouter>
  );
  await screen.findByDisplayValue('mount hood');

  expect(container).toMatchSnapshot();
});
