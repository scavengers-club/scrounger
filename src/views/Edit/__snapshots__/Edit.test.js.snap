// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit should render prefilled form  1`] = `
<div>
  <div>
    <form
      class="adventure-form"
    >
      <label>
        name:
        <input
          placeholder="name"
          required=""
          type="text"
          value="mount hood"
        />
      </label>
      <label>
        image:
        <input
          alt="mount hood"
          type="file"
        />
      </label>
      <img
        src="https://kfnfolzoauwbouvhqbrv.supabase.in/storage/v1/object/public/adv-images/trillium.jpg"
      />
      <label>
        description:
        <textarea
          placeholder="description"
          required=""
          type="text"
        >
          this adventure is for people, but also animals
        </textarea>
      </label>
      <label>
        hint 1:
        <textarea
          placeholder="hint 1"
          required=""
          type="text"
        >
          hint 1
        </textarea>
      </label>
      <label>
        hint 2:
        <textarea
          placeholder="hint 2"
          required=""
          type="text"
        >
          hint 2
        </textarea>
      </label>
      <label>
        hint 3:
        <textarea
          placeholder="hint 3"
          required=""
          type="text"
        >
          hint 3
        </textarea>
      </label>
      <label>
        solution:
        <input
          placeholder="solution"
          required=""
          type="text"
          value="Trillium Lake"
        />
      </label>
      <button>
        Submit
      </button>
    </form>
  </div>
   
</div>
`;
